mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -868,3 +907,93 @@ impl ChecksumConfig {
         &self.inner
     }
 }
+
+/// Checksum algorithm.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ChecksumAlgorithm {
+    /// No checksums.
+    None,
+    /// Crc32c checksum.
+    Crc32c,
+    /// Crc32 checksum.
+    Crc32,
+    /// Sha1 checksum.
+    Sha1,
+    /// Sha256 checksum.
+    Sha256,
+}
+
+impl From<aws_s3_checksum_algorithm> for ChecksumAlgorithm {
+    fn from(algorithm: aws_s3_checksum_algorithm) -> Self {
+        match algorithm {
+            aws_s3_checksum_algorithm::AWS_SCA_NONE => ChecksumAlgorithm::None,
+            aws_s3_checksum_algorithm::AWS_SCA_CRC32C => ChecksumAlgorithm::Crc32c,
+            aws_s3_checksum_algorithm::AWS_SCA_CRC32 => ChecksumAlgorithm::Crc32,
+            aws_s3_checksum_algorithm::AWS_SCA_SHA1 => ChecksumAlgorithm::Sha1,
+            aws_s3_checksum_algorithm::AWS_SCA_SHA256 => ChecksumAlgorithm::Sha256,
+            _ => unreachable!("unknown aws_s3_checksum_algorithm"),
+        }
+    }
+}
+
+impl From<ChecksumAlgorithm> for aws_s3_checksum_algorithm {
+    fn from(algorithm: ChecksumAlgorithm) -> Self {
+        match algorithm {
+            ChecksumAlgorithm::None => aws_s3_checksum_algorithm::AWS_SCA_NONE,
+            ChecksumAlgorithm::Crc32c => aws_s3_checksum_algorithm::AWS_SCA_CRC32C,
+            ChecksumAlgorithm::Crc32 => aws_s3_checksum_algorithm::AWS_SCA_CRC32,
+            ChecksumAlgorithm::Sha1 => aws_s3_checksum_algorithm::AWS_SCA_SHA1,
+            ChecksumAlgorithm::Sha256 => aws_s3_checksum_algorithm::AWS_SCA_SHA256,
+        }
+    }
+}
+
+/// Info for the caller to review before an upload completes.
+#[derive(Debug)]
+pub struct UploadReview {
+    /// Info about each part uploaded.
+    pub parts: Vec<UploadPartReview>,
+    /// The checksum algorithm used.
+    pub checksum_algorithm: ChecksumAlgorithm,
```

Comment:
Maybe slightly more idiomatic if this is `Option<ChecksumAlgorithm>` rather than literally copying the `None` variant?